### PR TITLE
Gradual Research Part Two

### PR DESCRIPTION
**Pull request by Geevies**
* Research is now contained globally, instead of per console, servers can still be used to save data.
* Doppler arrays now report explosions to the science channel.
* Added slime cells to xenobiology.
* Scanner goggles now gives some info on slimes.
* Explosions on the same level as a doppler array, tesla coils set to produce research, suduko, and slime cells add research points to specific techs.
* Shooting any living being in a firing range will now produce research points, based on how much damage is dealt.

Explosions give: Material and Phoron
Slime Cells give: Bluespace, Biological, Magnet
Firing Ranges give: Combat, Biological
Winning Sudoku gives: Data
Tesla Coils give: Engineering, Power

I'm not sure about the numbers for all of these, but playtesting showed that they're relatively sane, but they can be tweaked after playtesting.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: vueui/src/components/view/mcomputer/research/researchtracker.vue [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      while (newValue >= 1000) {
+        newValue /= 1000;
+        suffixNum++;
+      }
+
+      newValue = Math.round(newValue);
+
+      newValue += suffixes[suffixNum];
+      return newValue;
+    }
```
Comment (Karolis2011) [inline vueui/src/components/view/mcomputer/research/researchtracker.vue:35]: It's better to do this as a filter - https://vuejs.org/v2/guide/filters.html

File: vueui/src/components/view/mcomputer/research/researchtracker.vue [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<template>
+  <div>
+    <h3>Research Levels:</h3>
+    <vui-group>
+      <vui-group-item v-for="tech in techs" :key="tech" :label="tech.name" v-if="tech.level > 1 || tech.next_level_progress > 0">
```
Comment (Karolis2011) [inline vueui/src/components/view/mcomputer/research/researchtracker.vue:5]:
```js
computed: {
  filterdTechs() {
    return this.techs.filter(t => t.level > 1 || t.next_level_progress > 0)
  }
}

**Comment by NonQueueingMatt** [comment]
Map conflicts, changes required, etc.

**Pull request closed without merging**